### PR TITLE
Document monitoring and metrics support

### PR DESCRIPTION
**Pull request by defagos**
# Description

This PR documents monitoring and metrics support.

# Changes made

- Add suffixes to articles and extensions to avoid undesired conflicts. Suffixes are not required for tutorials and outlines (`tutorial` extension).
- Improve metric event naming consistency (metadata / asset). This agrees not only with our general naming but also with conventions we apply in our monitoring JSON exchange format.
- Organize APIs in top-level documentation files.
- Fix SwiftUI extension references for iOS / tvOS 18+. Warnings will be emitted for previous versions but version 18 is around the corner anyway.
- Add default values for metrics tracker configuration parameters.
- Add privacy manifest to the monitoring package.

# Checklist

- [x] APIs have been properly documented (if relevant).
- [x] The documentation has been updated (if relevant).
- [x] New unit tests have been written (if relevant).
- [x] The demo has been updated (if relevant).

### CODE REVIEW

File: Sources/Player/Player.docc/Articles/metrics/metrics-article.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+Subscribe to ``Player/periodicMetricsPublisher(forInterval:queue:limit:)`` to receive a stream of ``Metrics`` related to the item currently being played.
+
+These metrics contain various instantaneous measurements, e.g. a bandwidth estimate or the bitrate of the content being played, but also cumulative values and increments since the last measurement. They are especially useful to understand not only what happens globally during a playback session, but also at regular time intervals.
+
+## Receive important metric-related events
+
+Subscribe to ``Player/metricEventsPublisher`` to receive a stream of ``MetricEvent``s related to the item currently being played.
+
+Events include timing events, most notably metadata and asset loading, as well as fatal and non-fatal errors. Please refer to the documentation to learn more about all ``MetricEvent/Kind-swift.enum``s of events that are currently supported.
```
Comment (waliid) [inline Sources/Player/Player.docc/Articles/metrics/metrics-article.md:34]:
```suggestion
Events include timing events, most notably metadata and asset loading, as well as warnings and errors. Please refer to the documentation to learn more about all ``MetricEvent/Kind-swift.enum``s of events that are currently supported.
```

In the code, we are using the term **warning** and not **non-fatal**. Isn't it easier to keep using these terms?

File: Sources/Player/Player.docc/Articles/tracking/tracking-article.md [hunk 2]
```diff
@@ -30,16 +30,17 @@ Once types associated with an item tracker have been defined, start implementing
...
 4. ``PlayerItemTracker/updateProperties(to:)`` is called when player properties change. Be careful that properties can change often and that your implementation should be as efficient as possible.
-5. ``PlayerItemTracker/disable(with:)`` is called when the player item stops being the current one. Your implementation should cleanup resources acquired in ``PlayerItemTracker/enable(for:)``. The provided properties can be used to extract player state information at the end of playback.
-6. Since item trackers are required to be classes you can use `deinit` to perform any necessary final cleanup when the tracker and its item are discarded.
+5. ``PlayerItemTracker/updateMetricEvents(to:)`` is called when a new ``MetricEvent`` has been received.
+6. ``PlayerItemTracker/disable(with:)`` is called when the player item stops being the current one. Your implementation should cleanup resources acquired in ``PlayerItemTracker/enable(for:)``. The provided properties can be used to extract player state information at the end of playback.
```
Comment (waliid) [inline Sources/Player/Player.docc/Articles/tracking/tracking-article.md:34]:
```suggestion
6. ``PlayerItemTracker/disable(with:)`` is called when the player item stops being the current one. Your implementation should clean up resources acquired in ``PlayerItemTracker/enable(for:)``. The provided properties can be used to extract player state information at the end of playback.
```

File: Sources/Player/Player.docc/Articles/tracking/tracking-article.md [hunk 3]
```diff
@@ -54,3 +55,12 @@ let item = PlayerItem.simple(url: url, metadata: CustomMetadata(), trackerAdapte
...
+
+### Associate a session identifier
+
+A ``PlayerItemTracker`` can associate an optional ``PlayerItemTracker/sessionIdentifier-329qf`` with content being played.
+
+Session identifiers are useful for trackers collecting data and sending it to an external service. In such cases you might namely want to match data received remotely with a local session. If this is the case:
+
+1. Implement ``PlayerItemTracker/sessionIdentifier-329qf`` and return an identifier for the current session. You might want to associate a different session with each playback attempt.
+2. When you need to obtain the list of identifiers in use for some type of tracker, call ``Player/currentSessionIdentifiers(trackedBy:)``, passing it the tracker type. How you make it possible to extract session identifiers client-side is entirely up to you. You might want to display them alongside error messages, automatically capture them into the clipboard, or add them to an email sent to some support recipient address, for example.
```
Comment (waliid) [inline Sources/Player/Player.docc/Articles/tracking/tracking-article.md:66]:
```suggestion
2. When you need to obtain the list of identifiers in use for some type of tracker, call ``Player/currentSessionIdentifiers(trackedBy:)``, passing it the tracker type. How you make it possible to extract session identifiers client-side is entirely up to you. You might want to display them alongside error messages, automatically capture them to the clipboard, or add them to an email sent to some support recipient address, for example.
```

**Review by waliid — APPROVED** [review]

File: Sources/Player/Player.docc/Articles/metrics/metrics-article.md (hunk 1 above)
Comment (defagos) [reply Sources/Player/Player.docc/Articles/metrics/metrics-article.md:34]: Good point

**Comment by defagos** [comment]
Thanks for the suggestions!

**Review by waliid — APPROVED** [review]

**Pull request merged**